### PR TITLE
Added I/O events sample.

### PR DESCRIPTION
**Pull request by GabrielRanghetti**
## Description

Added an I/O Event example that listens to a Commerce event, handles the data and send to a Slack channel.

## Related Issue

None

## Motivation and Context

Accelerates onboarding to I/O Events and improves developer experience.

## How Has This Been Tested?

Tested by triggering the Commerce event and check the slack channel for the data.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: events/README.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+The runtime action ```events/customer-login``` will receive the Commerce event payload and send to the configured Slack channel.
+
+### Commerce event payload example:
+
+```json
+{
+    "data": {
+        "key": "b2d1addf-50af-4ae9-8d29-26a786d5161f",
+        "value": {},
```
Comment (ryanjcohen) [inline events/README.md:34]: Minor comment, but maybe some comment/indication that `value` will contain the fields from the `observer.customer_login` event subscription would be useful just so there's no doubt for those without experience receiving Commerce events in a runtime action

File: events/actions/customer-login/index.js [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+     // post the message to external api endpoint
+     var slackText = "Customer Login - " + JSON.stringify(params)
+
+     const payload = {
+       "channel": params.SLACK_CHANNEL,
+       "username": "incoming-commerce-evergreen-event",
+       "text": slackText,
+       "mrkdwn": true
+     }
```
Comment (ryanjcohen) [inline events/actions/customer-login/index.js:56]: https://api.slack.com/messaging/webhooks#posting_with_webhooks shows an example where only the text is needed. Did the other key/value pairs come from a different doc somewhere? I tried making changes like setting the channel name to one that doesn't exist within the workspace, and the message still seems to get posted to the channel selected when configuring the webhook

Comment (GabrielRanghetti) [reply events/actions/customer-login/index.js:56]: @ryanjcohen This actually comes from a demo extension when we install the @adobe/commerce-events-ext-tpl template:
image

Comment (GabrielRanghetti) [reply events/actions/customer-login/index.js:56]: But you right, it seems it only needs the text field.

File: events/.eslintrc.json [hunk 3]
```diff
@@ -0,0 +1,10 @@
+{
+    "env": {
+        "es6": true,
+        "node": true
+    },
+    "extends": ["eslint:recommended", "plugin:jest/recommended"],
+    "parserOptions": {
+        "ecmaVersion": "latest"
+    }
+}
```
[the remark this answers is not in the document]
Comment (GabrielRanghetti) [reply events/.eslintrc.json]: Added empty line.

File: events/README.md [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+With Adobe I/O Events, developers can create event-driven applications that take action when a shopper performs an action on an Adobe product. In this example the app will listen to the Adobe Commerce ```observer.customer_login``` event, get the payload and send to a Slack channel.
+
+## Prerequisites
+
+Before you begin, ensure you have the following:
+
+- An Adobe Developer account
+- Node.js and npm installed on your local machine (nvm 18.x.x (Mac/Linux) or nvm-windows (Windows))
+- App Builder project created and configured to work with I/O Events.
+- A Slack App configured to receive incoming webhooks.
```
[the remark this answers is not in the document]
Comment (GabrielRanghetti) [reply events/README.md]: Added Slack docs link.

File: events/README.md (hunk 1 above)
Comment (GabrielRanghetti) [reply events/README.md]: Updated README file.

File: events/README.md [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+
+### Commerce configured io_events.xml file:
+
+```xml
+<config xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+    xsi:noNamespaceSchemaLocation="urn:magento:module:Magento_AdobeCommerceEventsClient:etc/io_events.xsd">
+    <event name="observer.customer_login">
+        <fields>
+            <field name="firstname" />
+            <field name="lastname" />
```
Comment (oshmyheliuk) [inline events/README.md:36]: The example of the event payload for `observer.customer_login` event:
```json
{
    "customer": {
        "entity_id": "16",
        "group_id": 1,
        "created_at": "2024-04-23 14:13:32",
        "updated_at": "2024-06-04 14:24:32",
        "created_in": "Default Store View",
        "email": "test13@test.com",
        "firstname": "test",
        "lastname": "test",
        "store_id": 1,
        "website_id": 1,
        "addresses": [],
        "disable_auto_group_change": 0,
        "custom_attributes": [
            {
                "attribute_code": "reward_update_notification",
                "value": "1"
            }
        ],
        "reward_update_notification": "1"
    }
}
```


As you can see the `firstname` and `lastname` fields are in the `customer` section. So the event subscription should have the following format:

```xml
<event name="observer.customer_login">
        <fields>
            <field name="customer.firstname" />
            <field name="customer.lastname" />
        </fields>
</event>
```

Comment (GabrielRanghetti) [reply events/README.md]: @oshmyheliuk Good point, I retested the integration and updated the README file.

**Review by oshmyheliuk — APPROVED** [review]

**Review by ryanjcohen — APPROVED** [review]

**Comment by slamech** [comment]
This sample looks like the /action folder is directly under /events. Do we need to create folders for each sample under events and move it under there?

**Pull request merged**